src/AI/WallyAI.java

```diff
@@ -48,8 +48,21 @@ public String getName()
     {
       return "Wally";
     }
+
+    @Override
+    public String getDescription()
+    {
+      return
+          "Wally thinks he's clever; he knows about artillery, and tries to keep his important units safe until he wants to engage.\n" +
+          "He's a little slow on the uptake, though.";
```

Comment:
I feel like your class-level comment was a bit more informative; I adapted that when I did my combined branch.
          "Wally values units based on firepower and the area they can threaten.\n" +
          "He tries to keep units out of harm's way, and to protect expensive units with cheaper ones.";

Calling out his love for artillery is probably a good thing too, but "slow on the uptake" isn't going to mean much to anyone.

Reply:
Yeah, this is true. I was trying to avoid getting dug into the weeds, but I think you did a better job of also being informative.

Reply:
I do have a vague plan of how to fix the timidity problem: Set up the threat-area-check to just return a list of threats per square, rather than an aggregate threat value.
That'd let Wally say "oh, this enemy is the only threat here, so I can just shoot it and not worry about reprisal"

The ugly bit is that now you have to iterate weapons and check attack/move ranges for every threatened square you check for each unit, since you have no way of knowing which weapons will reach and how much damage they'll do.
I guess this is my due recompense for overengineering the weapon system, hah.